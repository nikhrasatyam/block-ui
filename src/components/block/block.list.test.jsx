import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import Enzyme from 'enzyme';
import UiStore from '../../stores/ui.store';
import BlockStore from '../../stores/block.store';
import blockListMock from '../../mocks/blockList.mock';

import BlockList from './block.list';

describe('BlockList Test Cases', () => {
  let uiStore;
  let blockStore;
  beforeEach(async () => {
    uiStore = new UiStore();
    blockStore = new BlockStore();
  });
  afterEach(() => { uiStore = null; blockStore = null; });

  test('should render BlockList when list is empty', () => {
    const tree = renderer.create(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render BlockList when list has data', () => {
    blockStore.setBlockList(blockListMock);
    const tree = renderer.create(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should display table and button when list is not empty', async () => {
    blockStore.setBlockList(blockListMock);
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>);
    expect(wrapper.find('Button')).toHaveLength(1);
    expect(wrapper.find('TableExpandable')).toHaveLength(1);
  });

  test('should not display table and button when list is empty', async () => {
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>);
    expect(wrapper.find('Button')).toHaveLength(0);
    expect(wrapper.find('TableExpandable')).toHaveLength(0);
  });

  test('should  display table and button when list is not empty', async () => {
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>);
    expect(wrapper.find('Button')).toHaveLength(0);
    expect(wrapper.find('TableExpandable')).toHaveLength(0);
  });

  test('should set the loading button disabled when showLoader is truthy', async () => {
    blockStore.setBlockList(blockListMock);
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>);
    let button = wrapper.find('Button');
    expect(button).toHaveLength(1);
    expect(button.props().disabled).toBeFalsy();
    uiStore.setShowLoader(true);
    wrapper.update();
    button = wrapper.find('Button');
    expect(button.props().disabled).toBeTruthy();
  });

  test('should update the blockStore when loading button is clicked', async () => {
    blockStore.setBlockList(blockListMock);
    const wrapper = Enzyme.mount(<Provider uiStore={uiStore} blockStore={blockStore}><BlockList /></Provider>);
    const button = wrapper.find('Button');
    expect(button).toHaveLength(1);
    expect(button.props().disabled).toBeFalsy();
    button.simulate('click');
    expect(blockStore.apiType.getBlockList).toBeTruthy();
    expect(blockStore.apiType.getAction).toBeTruthy();
  });
});
