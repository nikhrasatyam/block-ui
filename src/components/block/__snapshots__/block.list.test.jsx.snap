// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Test Cases should render BlockList when list has data 1`] = `
Array [
  <button
    className="ui primary button"
    onClick={[Function]}
  >
    Load
  </button>,
  <div
    className="sc-bdVaJa fePDys"
  >
    <div
      className="react-bootstrap-table"
    >
      <table
        className="table table-bordered"
      >
        <thead>
          <tr>
            <th
              className="sortable"
              onClick={[Function]}
              style={
                Object {
                  "textAlign": "center",
                  "width": "50%",
                }
              }
              tabIndex={0}
            >
              Id
              <span
                className="order"
              >
                <span
                  className="dropdown"
                >
                  <span
                    className="caret"
                  />
                </span>
                <span
                  className="dropup"
                >
                  <span
                    className="caret"
                  />
                </span>
              </span>
            </th>
            <th
              className="sortable"
              onClick={[Function]}
              tabIndex={0}
            >
              Timestamp
              <span
                className="order"
              >
                <span
                  className="dropdown"
                >
                  <span
                    className="caret"
                  />
                </span>
                <span
                  className="dropup"
                >
                  <span
                    className="caret"
                  />
                </span>
              </span>
            </th>
            <th
              className="sortable"
              onClick={[Function]}
              tabIndex={0}
            >
              Actions
              <span
                className="order"
              >
                <span
                  className="dropdown"
                >
                  <span
                    className="caret"
                  />
                </span>
                <span
                  className="dropup"
                >
                  <span
                    className="caret"
                  />
                </span>
              </span>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
            onClick={[Function]}
            style={Object {}}
          >
            <td>
              0585a09436e9b458890f59a35dceb202847b0944e76718ff768fe09938547c79
            </td>
            <td>
              2019-11-30T23:56:43.500
            </td>
            <td>
              28
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
]
`;

exports[`Alert Test Cases should render BlockList when list is empty 1`] = `null`;
